internal/users/locking/locking.go

```diff
@@ -30,7 +31,21 @@ var (
 // database in write mode, while it will return an error if called while the
 // lock is already hold by this process.
 func WriteLock() error {
-	return writeLockImpl()
+	done := make(chan error)
+	go func() {
+		done <- writeLockImpl()
+	}()
+
+	select {
+	// lckpwdf when called from gco doesn't behave exactly the same, likely
+	// because alarms are handled by go runtime, so do it manually here by
+	// failing if "lock not obtained within 15 seconds" as per lckpwdf.3.
+	// Keep this in sync with what lckpwdf does, adding an extra second.
+	case <-time.After(16 * time.Second):
```

Comment:
We might want to use a variable here which can be overridden in the tests, so that the test doesn't take 16 seconds to run.

Reply:
If this is only testing in non integration tests to trigger an error, I think this is a good candidate for synctest (https://go.dev/blog/synctest), which is behind a feature flag for 1.24: `GOEXPERIMENT=synctest`. Mind giving it a try?

Reply:
Marco and I discussed this briefly in a HO. It's a nice feature and will be useful in the future, but for this specific package we don't gain much by reducing the runtime of the test, because we have [this other test](https://github.com/ubuntu/authd/blob/7b5640b88916a2c3eddcbb6bccdb0ee94f9883b2/internal/users/locking/locking_bwrap_test.go#L21-L21) which calls `gpasswd` to test that it times out after 15 seconds when we hold the lock. We assume `synctest` won't help us speeding up the sleeps of subprocesses, so running the tests of this package will take ~15 seconds anyway.